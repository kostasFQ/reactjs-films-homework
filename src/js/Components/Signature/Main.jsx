import { hot } from 'react-hot-loader';
import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.scss';

const Main = (props) => {
  const { name } = props;
  return (
    <div className="content">
      <h1 className={styles.name}>{name}</h1>
    </div>
  );
};

Main.propTypes = {
  name: PropTypes.string.isRequired,
};
export default hot(module)(Main);
