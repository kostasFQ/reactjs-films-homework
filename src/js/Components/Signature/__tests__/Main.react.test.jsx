import React from 'react';
import renderer from 'react-test-renderer';
import Main from '../Main';

test('test component', () => {
  const render = renderer.create(
    <Main name="KST" />,
  );

  const component = render.toJSON();

  expect(component).toMatchSnapshot();
  expect(component.type).toBe('div');
  expect.objectContaining({
    className: 'content',
  });
  expect.objectContaining({
    children: expect.any(Object),
  });
});
