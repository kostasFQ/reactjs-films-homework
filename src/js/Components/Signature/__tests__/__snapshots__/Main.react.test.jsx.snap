// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component 1`] = `
<div
  className="content"
>
  <h1
    className="name"
  >
    KST
  </h1>
</div>
`;
